<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Creating non-canonical amino acid ChemComp files</title>
    <link rel="stylesheet" href="CcpnVideoTutorials.css">
</head>
<body>
<div class="header"><a href="https://www.ccpn.ac.uk/"><img src="CCPN.png" alt="CCPN Logo" height="50"></a><a
        href="index.html" class="button">Video<br> Tutorials<br> & Manual</a><a href="https://forums.ccpn.ac.uk"><img
        src="CCPNForumLogo.png" alt="CCPN Logo" height="50" style="padding-left: 15px"></a></div>
<br>
<div class="headerBar">Creating non-canonical amino acid ChemComp files</div>
<div class="tutorial">

    <div class="videoContainer">
        <div class="videoWrapper">
            <iframe src="https://www.youtube.com/embed/G6p_JFn9ZQc" frameborder="0"
                    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                    allowfullscreen></iframe>
        </div>
    </div>

    <h2>Creating non-canonical amino acid ChemComp files</h2>
    <h3>ChemComp files</h3>
    <p>If you want to include an amino acid in a protein/peptide Chain in CcpNmr Analysis which is not one of the 20 standard natural amino acids, you will need a so-called ChemComp file for this amino acid. A ChemComp file is an .xml file which contains all the chemical information about your amino acid: atoms, bonds, protonation states, names, codes and information about magnetic equivalence, prochirality etc. The ChemComp files for the 20 natural amino acids as well as a some other common derivatives (e.g. the D-amino acids) are provided as part of your CcpNmr Analysis Version 3 distribution (see below for a <a href="#AnalysisAAChemComps">full list of these</a>). The .xml files themselves are stored in your Analysis installation path in the <span class="action">ccpnmr3.x.y/src/python/ccpnmodel/data/ccpnv3/ccp/molecule/ChemComp</span> folder.</p>
    <p>Additionally, you can find over 3000 other ChemComp files for non-natural amino acids on the <a href="https://github.com/VuisterLab">VuisterLab GitHub page</a> at <a href="https://github.com/VuisterLab/CcpNmr-ChemComps/tree/master/data/pdbe/chemComp/archive/ChemComp/protein">https://github.com/VuisterLab/CcpNmr-ChemComps/tree/master/data/pdbe/chemComp/archive/ChemComp/protein</a>. You can download any of these if you need them for your peptide/protein Chain. This repository is not very user-friendly when it comes to searching for your desired molecule. We recommend using the PDB's <a href="https://www4.rcsb.org/chemical-sketch">Chemical Sketch Tool</a> to find your molecule of interest in the PDB. The PDB code (which we refer to as the chemComp code) should be 3-5 letters in length. You can then search for the code in the search box in the left-hand sidebar in our <a href="https://github.com/VuisterLab/CcpNmr-ChemComps/tree/master/data/pdbe/chemComp/archive/ChemComp/protein">GitHub repository</a> to see if it is present.</p>
    <p>If you can't find a ChemComp for the amino acid you require, you can create your own using ChemBuild.</p>

    <h3>ChemBuild</h3>
    <p>ChemBuild is a program written by Tim Stevens which is part of the CcpNmr suite of programs. The executable for this program is located in your <span class="action">ccpnmr3.x.y/bat</span> (Windows) or <span class="action">ccpnmr3.x.y/bin</span> (Linux/Mac) folder along with the other CcpNmr Analysis executables. To start the program simply double-click <span class="action">ChemBuild.bat</span> (Windows) or <span class="action">chemBuild</span> (Linux/Mac) in your file browser or use the commandline argument <span class="action">./chemBuild</span> in the <span class="action">bin</span> folder (Linux/Mac).</p>
    <p>ChemBuild is similar to many other chemical drawing programs, but it also adds information about magnetic equivalence and is particularly good at dealing with peptides/proteins composed of linked amino acids (or similarly polynucleotides). It also has the ability to read and write CCPN ChemComps files. A <a>manual</a> is available, but much of what you will need to know to create an amino acid ChemComp is either relatively intuitive or will be described here.</p>

    <h3>Creating your own bespoke ChemComp</h3>
    <h4>Linked Amino Acid Templates</h4>
    <p>In the <span class="action">Compound Library</span> on the left-hand side of the ChemBuild Desktop, opne up the <span class="action">Amino Acids</span> section and then drag the <span class="action">Linked Template</span> onto the main canvas. On the right-hand side, select the <span class="action">Compound Variants</span> tab. You will see that the amino acid template contains a number of different <span class="action">Compound Variants</span>, often with different pronation states. Some variants are of the free peptide, others of the amino acid at the N-terminal position in a chain (labelled <span class="action">start</span>), a middle positoin (labelled <span class="action">middle</span>) and others at the C-terminal position (labelled <span class="action">end</span>). The linked variants will show little green arrows on the molecule where they link to neighbouring amino acids. Select the <span class="action">middle</span> variant while you build your amino acid, so that you don't accidentally remove the links (this is particularly important if you will be replacing the amide hydrogen with a different chemical group.</p>
    <h4>Building your Amino Acid</h4>
    <p>You can build your amino acid by selecting, adding or deleting atoms and bonds.</p>
    <p><span class="action">Select</span> atoms by clicking on them. Hold down the <span class="action">Ctrl</span> key (the <span class="action">Cmd</span> key on a Mac) to select multiple atoms in one go. You can also right-drag the mouse over multiple atoms to select several in one go. And finally, you can use <span class="action">Ctrl/Cmd+A</span> to select all atoms.</p>
    <p><span class="action">Delete</span> selected atoms by pressing the <span class="action">Delete</span> button (or <span class="action">Fn+Backspace</span> on a Mac). Alternatively use the main menu (<span class="action">Edit / Delete atoms</span>), right-hand mouse menu (<span class="action">Edit / Delete selected atoms</span>) or use the toolbar (icon with the black atom with a red cross over it).</p>
    <p><span class="action">Add</span> atoms by dragging them from the <span class="action">Elements</span> section on the left-hand side or by pressing the letter keys for atoms with one-letter atomic symbols (<span class="action">C</span> for carbon etc.). If you press <span class="action">C</span> while another atom is selected, the new carbon atom will automatically be bonded to the selected atom, otherwise your new atom will be placed onto the canvas without any bonds. Note that each element contains black dots for any free valencies. You can automatically add hydrogen atoms to all remaining free valencies of your selected atoms by using the menus (<span class="action">Edit / Add Hydrogens</span>) or toolbar button (green plus icon with two small grey H atoms attached).</p>
    <p><span class="action">Create bonds</span> between atoms by dragging one atom towards another, such that two valency dots are close and become highlighted in green. Let go of the dragged atom and a bond will form between the two atoms.</p>
    <p><span class="action">Delete bonds</span> by deleting atoms, or by selecting atoms and then using the menus (<span class="action">Edit / Remove bonds</span>) or toolbar (icon with two bonded atoms and a red cross over the bond). </p>
    <p><span class="action">Rearranging</span> atoms is possible by dragging any selected atoms with the mouse (if several atoms ar selected they are dragged as a group). You can also use several buttons in the toolbar: the purple circular arrows will rotate the selected atom (group), the next two buttons with flip the atom (group) vertically or horizontally and the button with three atoms with green arrows between them will auto-arrange your atoms. These functions are also all available from the <span class="action">Edit</span> menu.</p>
    <p>You can also make use of the <span class="action">Compound Library</span> on the left-hand side of the Desktop to drag particular chemical groups onto the canvas and connect them to your amino acid template.</p>
    <h4>Importing Molecules</h4>
    <p>It is also possible to import molecules in .pdb, .mdl or .mol2 format. Go to the <span class="action">Import</span> menu and select the format you wish to import. You can then connect up your imported molecule to the amino acid template. But it is always important to start with the template, as you need the differently connected variants, and it isn't currently possible to create these yourself <i>ab initio</i> in ChemBuild.</p>
    <h4>Changing Labels</h4>
    <p>You may have noticed that the linked template starts with the standard C/N/CA/CB labelling for an amino acid. Additional groups will simply be lablled with numbers (C1, C2, H1 etc.). We suggest you adapt the atom labels to fit in with the usual Greek alphabet labelling. To change a label, simply double-click on a label, change it as desired and press <span class="action">Enter</span>.</p>
    <h4>Adding Magnetic Properties</h4>
    <p>You may have noticed that the hydrogen atoms of a methyl group are automatically connected via red dashed lines. This indicates that these atoms are magnetically equivalent. Other sets of atoms (e.g. the hydrogen atoms in a CH<sub>2</sub> group) may be connected by blue dashed lines. This indicates sets of prochiral atoms. Most of these groups will be recognised automatically, but phenyl rings typically need to have the magnetic equivalence added manually. To do this, select the <span class="action">Atom Properties</span> tab on the right-hand side of the ChemBuild Desktop. Select each of the pairs of carbon or hydrogen atoms that are magnetically equivalent in turn and then press the <span class="action">Set equivalents</span> button in the <span class="action">NMR Groups</span> section. For each pair of equivalent atoms you also need to make sure that the atom names match each other (e.g. CH1 and CH2, HZ1 and HZ2, C16_1 and C16_2, H74 and H75), i.e.they always need a root that is the same and then they need to end with single-digit numbers which differ.</p>
    <h4>Compound Information</h4>
    <p>Before exporting your amino acid in ChemComp format, you need to edit the Compound Information. Select <span class="action">Compound Information</span> on the right-hand side of the ChemBuild Desktop. Enter a name for your compound and set the <span class="action">CCPN MolType</span> to <span class="action">protein</span>. The CCPN Code can either be left as it is (the code is a random and almost certainly unique code), or you can enter a three-letter code. But make sure this does not clash with one of the three-letter codes used in the list of amino acids on <a href="https://github.com/VuisterLab/CcpNmr-ChemComps/tree/master/data/pdbe/chemComp/archive/ChemComp/protein">GitHub</a> or else you won't be able to use your amino acid within CcpNmr Analysis. If you want to use your ChemComp within CcpNmr Analysis Version 2, then you will need to use a three-letter code.</p>
    <h4>Exporting the ChemComp .xml file</h4>
    <p>Once your amino acid is ready for export, go to the <span class="action">Export</span> menu and select <span class="action">CCPN ChemComp XML file</span> and select a distination. The folder will be saved with the .xml file inside it.</p>

    <h3 id="AnalysisAAChemComps">Amino Acid ChemComp files included in the CcpNmr Analysis Version 3 Distribution</h3>
    <table>
    <tr><th>3-Letter Code</th><th>Amino Acid name</th></tr>
<tr><td>Aba</td><td>(2R)-2-aminobutanoic acid</td></tr>
<tr><td>Aib</td><td>2-methylalanine</td></tr>
<tr><td>Ala</td><td>alanine</td></tr>
<tr><td>Arg</td><td>arginine</td></tr>
<tr><td>Asn</td><td>asparagine</td></tr>
<tr><td>Asp</td><td>aspartic acid</td></tr>
<tr><td>Bmt</td><td>4-methyl-4-[(E)-2-butenyl]-4,N-methyl-threonine</td></tr>
<tr><td>Cgu</td><td>gamma-carboxy-glutamic acid</td></tr>
<tr><td>Cso</td><td>S-hydroxycysteine</td></tr>
<tr><td>Css</td><td>S-mercaptocysteine</td></tr>
<tr><td>Cxm</td><td>N-carboxymethionine</td></tr>
<tr><td>Cys</td><td>cysteine</td></tr>
<tr><td>D11</td><td>D-phoshphothreonine</td></tr>
<tr><td>Dal</td><td>D-alanine</td></tr>
<tr><td>Dar</td><td>D-arginine</td></tr>
<tr><td>Das</td><td>D-aspartic acid</td></tr>
<tr><td>Dcy</td><td>D-cysteine</td></tr>
<tr><td>Dgl</td><td>D-glutamic acid</td></tr>
<tr><td>Dgn</td><td>D-glutamine</td></tr>
<tr><td>Dha</td><td>2-aminoacrylic acid</td></tr>
<tr><td>Dhi</td><td>D-histidine</td></tr>
<tr><td>Dil</td><td>D-isoleucine</td></tr>
<tr><td>Div</td><td>D-isovaline</td></tr>
<tr><td>Dle</td><td>D-leucine</td></tr>
<tr><td>Dly</td><td>D-lysine</td></tr>
<tr><td>Dne</td><td>D-norleucine</td></tr>
<tr><td>Dng</td><td>N-formyl-D-norleucine</td></tr>
<tr><td>Dnm</td><td>N-methyl-D-norleucine</td></tr>
<tr><td>Dpn</td><td>D-phenylalanine</td></tr>
<tr><td>Dpr</td><td>D-proline</td></tr>
<tr><td>Dse</td><td>N-methyl-D-serine</td></tr>
<tr><td>Dsg</td><td>D-asparagine</td></tr>
<tr><td>Dsn</td><td>D-serine</td></tr>
<tr><td>Dth</td><td>D-threonine</td></tr>
<tr><td>Dtr</td><td>D-tryptophan</td></tr>
<tr><td>Dty</td><td>D-tyrosine</td></tr>
<tr><td>Dva</td><td>D-valine</td></tr>
<tr><td>Gln</td><td>glutamine</td></tr>
<tr><td>Glu</td><td>glutamic acid</td></tr>
<tr><td>Gly</td><td>glycine</td></tr>
<tr><td>His</td><td>L-histidine</td></tr>
<tr><td>Hsl</td><td>homoserine lactone</td></tr>
<tr><td>Hyp</td><td>4-hydroxyproline</td></tr>
<tr><td>Ile</td><td>isoleucine</td></tr>
<tr><td>Ioy</td><td>P-iodo-D-phenylalanine</td></tr>
<tr><td>Iyt</td><td>N-alpha-acetyl-3,5-diiodotyrosyl-D-threonine</td></tr>
<tr><td>Kcx</td><td>lysine NZ-carboxylic acid</td></tr>
<tr><td>Leu</td><td>leucine</td></tr>
<tr><td>Llp</td><td>2-lysine(3-hydroxy-2-methyl-5-phosphonooxymethyl-pyridin-4-ylmethane)</td></tr>
<tr><td>Lys</td><td>lysine</td></tr>
<tr><td>Med</td><td>D-methionine</td></tr>
<tr><td>Met</td><td>methionine</td></tr>
<tr><td>Mle</td><td>N-methylleucine</td></tr>
<tr><td>Mse</td><td>selenomethionine</td></tr>
<tr><td>Mva</td><td>N-methylvaline</td></tr>
<tr><td>Ndf</td><td>N-(carboxycarbonyl)-D-phenylalanine</td></tr>
<tr><td>Nle</td><td>norleucine</td></tr>
<tr><td>Ocs</td><td>cysteinesulfonic acid</td></tr>
<tr><td>Orn</td><td>ornithine</td></tr>
<tr><td>Pca</td><td>pyroglutamic acid</td></tr>
<tr><td>Pdd</td><td>N-(5'-phosphopyridoxyl)-D-alanine</td></tr>
<tr><td>Pg9</td><td>D-phenylglycine</td></tr>
<tr><td>Phe</td><td>phenylalanine</td></tr>
<tr><td>Pro</td><td>proline</td></tr>
<tr><td>Ptr</td><td>O-phosphotyrosine</td></tr>
<tr><td>Sar</td><td>sarcosine</td></tr>
<tr><td>Ser</td><td>serine</td></tr>
<tr><td>Thr</td><td>threonine</td></tr>
<tr><td>Trp</td><td>tryptophan</td></tr>
<tr><td>Tyr</td><td>tyrosine</td></tr>
<tr><td>Tys</td><td>sulfonated tyrosine</td></tr>
<tr><td>Val</td><td>valine</td></tr>
    </table>

</div>
</body>
</html>